src/Optimization/hiopNlpFormulation.cpp

```diff
@@ -1082,16 +1082,32 @@ void hiopNlpFormulation::user_callback_solution(hiopSolveStatus status,
   //! todo -> test this when fixed variables are removed -> the internal
   //! zl and zu may have different sizes than what user expects since HiOp removes
   //! variables internally
-  interface_base.solution_callback(status, 
-                                   (int)n_vars_,
-                                   x.local_data_const(),
-                                   z_L.local_data_const(),
-                                   z_U.local_data_const(),
-                                   (int)n_cons_,
-                                   cons_body_->local_data_const(),
-                                   cons_lambdas_->local_data_const(),
-                                   inv_obj_scale*obj_value);
-                                   
+  if(options->GetString("callback_mem_space")=="device") {
```

Comment:
see @maksud comment

Reply:
umm... it's github's problem. I did fix this issue in the last commit [9e8c8ab](https://github.com/LLNL/hiop/pull/535/commits/9e8c8ab1c64a04787519f6ad3ebbfb8f006e3ec6), but it seems that github didn't show it.
@cnpetra @maksud

Reply:
just push the change